install

```diff
@@ -233,18 +222,29 @@ abort <<-EOABORT unless !mac? || `dsmemberutil checkmembership -U "#{ENV["USER"]
 This script requires the user #{ENV["USER"]} to be an Administrator.
 EOABORT
 
+@have_sudo = false
 unless mac?
   if File.writable?(HOMEBREW_PREFIX_DEFAULT) || File.writable?("/home/linuxbrew") || File.writable?("/home")
     HOMEBREW_PREFIX = HOMEBREW_PREFIX_DEFAULT.freeze
   else
-    Kernel.system "/usr/bin/sudo -n -v 2>/dev/null"
-    unless $CHILD_STATUS.success?
+    io = IO.popen "/usr/bin/sudo -n /bin/mkdir -p #{HOMEBREW_PREFIX_DEFAULT}", :err => [:child, :out]
+    out = io.read
+    io.close
+    @have_sudo = $CHILD_STATUS.success?
+    if (not @have_sudo) && out == "sudo: a password is required\n"
```

Comment:
How about…
```ruby
    Kernel.system "/usr/bin/sudo -n -l /bin/mkdir >/dev/null 2>/dev/null"
    unless $CHILD_STATUS.success?
```

Reply:
Some users have right to do `sudo` with password, and some have right w/o password.
This is needed to distinguish users which do not have the right anyway.
Otherwise, the next prompt will be strange for these users.

Reply:
`sudo -l /bin/mkdir` instead of `sudo /bin/mkdir -p #{HOMEBREW_PREFIX_DEFAULT}` seems fine.

Reply:
You can use `sudo -nv` to check whether a user has `sudo` access without a password.

Reply:
Yes, but it fails if the user can do `sudo` only with a password.
Therefore, here, 

1. first check if the user can use `sudo` without password by `sudo -n -l mkdir`.
2. If failed, check if a password is required or not by the output.
3. Only if a password is required, ask user to put password to create /home/linuxbrew otherwise make $HOME/.linuxbrew

Reply:
I suggest…
```ruby
if `sudo -nv 2>&1` == "sudo: a password is required\n"
```

Reply:
Yes, that is enough in this case. updated.